apps/frontend/src/pages/quiz/libs/components/onboarding/onboarding.tsx

```diff
@@ -8,98 +7,83 @@ import {
 import styles from "./styles.module.css";
 
-const Onboarding: React.FC = () => {
-	const dispatch = useAppDispatch();
+type Properties = {
+	onNext: () => void;
+};
 
-	const [answerIds, setAnswerIds] = useState<OnboardingAnswerRequestBodyDto>({
-		answerIds: [],
-	});
+type FormToSave = {
+	[key: string]: string[];
+};
+
+const ZERO = 0;
+const ONE = 1;
+
+const OnboardingForm: React.FC<Properties> = ({ onNext }: Properties) => {
+	const dispatch = useAppDispatch();
+	const [isLastQuestion, setIsLastQuestion] = useState<boolean>(false);
 
-	const {
-		currentQuestionIndex,
-		isFirstQuestion,
-		isLastQuestion,
-		question,
-		totalQuestionsAmount,
-	} = useAppSelector(({ onboarding }) => ({
-		currentQuestionIndex: onboarding.currentQuestionIndex,
-		isFirstQuestion: onboarding.currentQuestionIndex === FIRST_QUESTION_INDEX,
-		isLastQuestion:
-			onboarding.currentQuestionIndex ===
-			onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
-		question: onboarding.currentQuestion,
-		totalQuestionsAmount: onboarding.questions.length,
-	}));
+	const { currentQuestionIndex, dataStatus, question, questions } =
+		useAppSelector(({ onboarding }) => ({
+			currentQuestionIndex: onboarding.currentQuestionIndex,
+			dataStatus: onboarding.dataStatus,
+			question: onboarding.currentQuestion,
+			questions: onboarding.questions,
+		}));
 
 	useEffect(() => {
 		void dispatch(onboardingActions.getAll());
 	}, [dispatch]);
 
-	const { control, handleSubmit, isValid, reset } =
-		useAppForm<OnboardingFormValues>({
-			defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
-			validationSchema: oneAnswerSelectedValidationSchema,
-		});
+	useEffect(() => {
+		if (
+			dataStatus === "fulfilled" &&
+			currentQuestionIndex === questions.length - ONE
+		) {
+			setIsLastQuestion(true);
+		}
+	}, [currentQuestionIndex, dataStatus, questions]);
+
+	const { control, handleSubmit } = useAppForm<OnboardingFormValues>({
+		defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
+	});
 
-	const handleSaveAnswers = useCallback(
-		(payload: OnboardingAnswerRequestBodyDto) => {
-			return payload; //TODO call api
-		},
-		[],
-	);
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
 
 	const handleNextStep = useCallback(
 		(data: OnboardingFormValues) => {
-			setAnswerIds((previousState) => {
-				const newAnswerIds = new Set([
-					...previousState.answerIds,
-					Number(data.answer),
-				]);
-
-				return {
-					answerIds: [...newAnswerIds],
-				};
-			});
-
-			if (isLastQuestion) {
-				handleSaveAnswers(answerIds);
-
-				return;
+			if (question) {
+				const hasAnswer = data[`answer${question.id.toString()}`] !== undefined;
+
+				if (hasAnswer) {
+					const newObject = Object.fromEntries(
+						Object.entries(data).filter(([key]) => key !== "answers"),
+					);
+
+					if (isLastQuestion) {
+						const answerIds = getAnswerIds(newObject);
```

Comment:
I was thinking more like keeping all ids in `answers` array, not creating a new key in object for each question, can we do that?